### DISCUSSION TITLE
Not able to assign second variable

### DISCUSSION THREAD

**Discussion by MTMishal** [nextgenhealthcare/connect#5188] [General]
hello All

I am trying simple DB query and trying it assign it to HL7 message. I am able to get data from DB. when i have simple JS in source transformer, I am able to assign the PID3 to correct filed. but when I assign PID5 (patient name ) or any other field. It is coming empty in encoded message. what could go wrong here 

here is the xml i receive after DB query. 

<result>
    <patientid>2203</patientid>
    <visitid>2203</visitid>
    <Name>testname</Name>
    <DOB>1940-02-14</DOB>
    <Sex>M</Sex>
    <Last_name>testlast</Last_name>
    <First_name>testfirst</First_name>
    <polling_Trigger>1</polling_Trigger>
</result>

and here is the 

tmp['PID']['PID.3']['PID.3.1'] = msg['patientid'].toString();
logger.info ("patient id received is" + msg['patientid'].toString());
tmp['PID']['PID.5']['PID.5.1'] = msg['last_name'].toString();
logger.info ("patient name received is" + msg['last_name'].toString());

**Comment by pacmano1** [comment]
```2203 2203 testname 1940-02-14 M testlast testfirst 1``` is not XML, post the real XML as shown in Mirth.

Please use code highlighting here also:

```JS
tmp['PID']['PID.3']['PID.3.1'] = msg['patientid'].toString();
logger.info ("patient id received is" + msg['patientid'].toString());
tmp['PID']['PID.5']['PID.5.1'] = msg['last_name'].toString();
logger.info ("patient name received is" + msg['last_name'].toString());
```
See https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks
